### ISSUE TITLE
Test Scenario: Batch Processing

### ISSUE THREAD

**Issue by jkva** [coyote-team/coyote-wp#101]
When a plugin has a working configuration, batch processing becomes available. The plugin should:

- process posts and pages;
  - in batches;
  - tracking progress using transients;
  - allow status updates and cancellation of a running job;
- use an optional process batch size parameter;
- process unpublished content based on a plugin setting; and
- ideally not use a remote processor as tracked by #95.

**Comment by Pen-a11y** [comment]
After running through each of the Test Scenarios for Batch Processing there are 2 of the scenarios that work fully and the rest will still need to be worked on for full functionality of the batch processor.
The following work correctly:
- use an optional process batch size parameter
- process unpublished content based on a plugin setting
- process posts and pages;
   - in batches;

The following functionality doesn't work correctly:
- process posts and pages;
   - tracking progress using transients: Currently, the tracking goes from 0% to 100%, and after testing with large numbers to import it does not seem to gradually increase the percentage while processing a batch.
   - allow status updates and cancellation of a running job: The cancellation of a running job works correctly, but the status update that is given is not very descriptive and when an error seems to occur the only message that is given is 'error'
 I think we should try to give a more descriptive error to the user, to tell the user what the issue could possibly be. 
- ideally not use a remote processor as tracked by https://github.com/coyote-team/coyote-wp/issues/95.
This issue has already been added earlier and will need to be addressed and talked about in there
